### ISSUE TITLE
iOS OpenGL error

### ISSUE THREAD

**Issue by paStafford** [googlevr/cardboard#241]
I'm working on an iOS test build.  When I run the app, the screen goes black and it just sits there.  Looking at the output in Xcode, I see an OpenGL error indicating a problem with the frame buffer.  I've included a snippet from the log below.  I'm using Unity 2019.4.17f1 LTS with OpenGLES 3.  My test phone is an iPhone 6 Plus (version 12.5.2).  Let me know if there's any other information I should provide.

RenderTexture->Create
RenderTexture->Create
IVRDeviceCallback_CreateEyeTextureResources 0x44
IVRDeviceCallback_CreateEyeTextureResources 0x45
OPENGL NATIVE PLUG-IN ERROR: GL_INVALID_FRAMEBUFFER_OPERATION: Framebuffer is not complete or incompatible with command 
(Filename: ./Runtime/GfxDevice/opengles/GfxDeviceGLES.cpp Line: 437)

2021-04-02 23:44:41.990643-0700 CardboardCarnivalGames[354:20288] GL error OpenGlEs3DistortionRenderer::RenderDistortionMesh: 1286
2021-04-02 23:44:41.991569-0700 CardboardCarnivalGames[354:20288] GL error OpenGlEs3DistortionRenderer::RenderDistortionMesh: 1286
2021-04-02 23:44:41.991820-0700 CardboardCarnivalGames[354:20288] [/Users/albachicar/localwork/release_v1_4_1/cardboard/sdk/unity/xr_unity_plugin/cardboard_xr_unity.cc : 95] [/Users/albachicar/localwork/release_v1_4_1/cardboard/sdk/unity/xr_unity_plugin/cardboard_xr_unity.cc : 624] GL error: 1286
(lldb)

**Comment by hawkwood** [comment]
Did you by any chance upgrade from a different version of Cardboard recently? I was getting this after switching from Mobfish's version to this, and I needed to deep clean my Unity project (quit, delete library, reimport new SDK, etc), then it went away.

**Comment by paStafford** [comment]
Thanks for the reply.  No, I've only used the Cardboard XR plugin v1.4.1 on this project.

After going through a multitude of builds trying to get something, anything to display on my test iPhone (using OpenGL), I believe the problem is Unity.  Without going through all the steps I tried along the way, I ended up making a clean build with out Cardboard or any of my code.  It was literally just the default sample scene with the graphics API set to OpenGL ES 3 and OpenGL ES 2.  Running it on the iPhone yields a black screen.  It doesn't even get to the Unity splash screen.

With Cardboard, I get the OpenGL errors in my original post.  Without Cardboard, I do not get any OpenGL errors in the console, just a black iPhone screen.  The last line in console output simply indicates OpenGL is deprecated on this platform.  Nothing in the console really jumps out at me as a clue to what's wrong.

Since I couldn't get a working build with my current version of Unity 2019.4.17f1 LTS, I downloaded the newest 2019.4.24fl LTS and gave that a try.  Same problems all the way around.

I don't know why, but it seems Unity 2019.4 simply will not allow me to make a non-Metal iOS build (I can make a Metal build, it just doesn't help me since Cardboard doesn't currently support Metal).  Regardless, this clearly isn't an issue with Cardboard, so I'll take this over to the Unity forums or StackOverflow.

**Comment by paStafford** [comment]
Thanks for the response.  Actually I've determined the issue isn't related to Cardboard at all.

After doing a variety of tests, I found I can't even build a normal, non-Cardboard iOS build that works.  The last build I tried was a brand new project using the default scene with no additions (no new assets, no packages, no code, etc).  When run the build on my test iPhone 6, it only produces a black screen (it doesn't even display the Unity splash screen).  The Xcode console doesn't show any errors in this setup.  Simply, the last line indicates OpenGL is deprecated on this platform.  

I tried the same setup in the newest version of 2019.4.x LTS (which is 2019.4.24f1) and got the same results.  With that last test, I should have ruled out any issues related to a corrupted installation.  I already knew OpenGL was deprecated and support would *eventually* be removed, but I was under the impression it was still supported.  I don't know if the issue is with Unity or iOS, but that's neither here nor there.  The problem isn't with Cardboard.  That said, it would seem I'm stuck waiting for the Metal update for Cardboard (I also did a test build with the test setup above using Metal and it worked fine).
